openassessment/xblock/submission_mixin.py

```diff
@@ -288,10 +327,46 @@ def create_submission(self, student_item_dict, student_sub_data):
 
         return submission
 
+    def collect_files_for_submission(self, student_sub_dict):
```

Comment:
I like that you made this a helper method.  Since it's meant to be "private" (i.e., only this class uses it directly), you should preface the method name with a `_`.